llm4pddl/utils.py

```diff
@@ -384,3 +385,33 @@ def str_to_identifier(x: str) -> str:
         https://stackoverflow.com/questions/5297448
     """
     return hashlib.md5(x.encode('utf-8')).hexdigest()
+
+
+def randomize_object_names(rng: np.random.Generator,
+                           objs: List[str]) -> Dict[str, str]:
+    """To prevent overfitting, creates dictionary mapping object names to
+    random strings."""
+    random_dict = {}
+    for obj in objs:
+        if obj not in random_dict:
+            random_dict[obj] = ''.join(
+                rng.choice(list(string.ascii_lowercase), len(obj)))
+    return random_dict
+
+
+def replace_with_random_objects(orig_str: str, random_dict: dict) -> str:
+    """Replaces objects in init, goal, solution string with random object
+    names."""
+    orig_str_array = orig_str.split("\n")
```

Comment:
it's possible that we won't always have new lines here. for example, if we use the `--llm_prompt_flatten_pddl` flag, then the new lines all get removed. i'm wondering if we can replace all occurrences of the substrings all at once, without explicitly looping? that's how we did it in the predicators implementation where we substituted option and predicate names -- you can take a look at that for reference